File: Native/Test/Python/get-member-bulk-data.py
#
#  Copyright 2010-2012 Fabric Technologies Inc. All rights reserved.
#

import fabric
F = fabric.createClient()

n = F.DG.createNode('n')
n.addMember('foo', 'String')
n.resize(3)
for i in range( 0, 3 ):
  n.setData('foo', i, "bar-"+str(i))

print(n.getMemberBulkData('foo'))
try:
  print(n.getMemberBulkData('bar'))
except Exception as e:
  print('Caught exception: ' + str(e))

print(n.getMembersBulkData(['foo']))
try:
  print(n.getMembersBulkData(['foo','bar']))
except Exception as e:
  print('Caught exception: ' + str(e))

F.close()

File: Native/Test/Node/get-member-bulk-data.js
/*
 *  Copyright 2010-2012 Fabric Engine Inc. All rights reserved.
 */

F = require('Fabric').createClient();

n = F.DG.createNode('n');
n.addMember('foo', 'String');
n.resize(3);
for (var i=0; i<3; ++i) {
  n.setData('foo', i, "bar-"+i);
}

console.log(n.getMemberBulkData('foo'));
try {
  console.log(n.getMemberBulkData('bar'));
}
catch (e) {
  console.log('Caught exception: ' + e);
}

console.log(n.getMembersBulkData(['foo']));
try {
  console.log(n.getMembersBulkData(['foo','bar']));
}
catch (e) {
  console.log('Caught exception: ' + e);
}

F.close();
